lib/discover/index.js

```diff
@@ -0,0 +1,21 @@
+var url = require('url')
+
+module.exports.protocols = {
+    "hyperbahn:": require('./hyperbahn').discover,
+    "file:": require('./file').discover,
+    "ip:": require('./ip').discover,
+    // by default fallback on ip discovery
+    default: require('./ip').discover
+}
+
+// find a list of initial nodes to connect to based on the discover string
+module.exports.discover = function (discover, callback) {
+    var urlObj = url.parse(discover)
+    var handler = module.exports.protocols[urlObj.protocol];
```

Comment:
Seems like a a switch statement would simplify this. I don't see any use of the exported `protocols` either:

```
switch (urlObj) {
case 'file:':
   discoveryByFile();
   break;
case 'hyperbahn:'
  discoverByHyperbahn();
  break
default:
  discoverByIP();
}
```